src/mat/4C_mat_micromaterialgp_static.hpp

```diff
@@ -108,6 +115,9 @@ namespace Mat
     /// my vector of new displacements
     std::shared_ptr<Core::LinAlg::Vector<double>> disn_;
 
+    /// history data for each column element on microscale
+    std::shared_ptr<std::map<int, std::shared_ptr<std::vector<char>>>> history_data_;
```

Comment:
This type is very weird. 

```suggestion
    std::map<int, std::vector<char>> history_data_;
```

Reply:
If you have many micro-elements, it might be better to use an unordered_map.

Reply:
Good point. Thanks.